grpc/sa-wrappers.go

```diff
@@ -189,6 +189,36 @@ func (sac StorageAuthorityClientWrapper) CountCertificatesByNames(ctx context.Co
 	return names, nil
 }
 
+func (sac StorageAuthorityClientWrapper) CountCertificatesByExactNames(ctx context.Context, domains []string, earliest, latest time.Time) (map[string]int, error) {
+	earliestNano := earliest.UnixNano()
+	latestNano := latest.UnixNano()
+
+	response, err := sac.inner.CountCertificatesByExactNames(ctx, &sapb.CountCertificatesByNamesRequest{
+		Names: domains,
+		Range: &sapb.Range{
+			Earliest: &earliestNano,
+			Latest:   &latestNano,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	if response == nil || response.CountByNames == nil {
+		return nil, errIncompleteResponse
+	}
+
+	names := make(map[string]int, len(response.CountByNames))
```

Comment:
Below, I talk about eliminating the map->list transform on the server side. What if we also eliminated the list->map transform on the client side? I.e. we would just return a list here, and the RA would iterate over a list rather than over a map. The map approach currently lets the RA check that the SA returned results for every name we asked for, but I think we can trust the SA to do that, or switch to a simpler length check. That would get us a little closer to killing our wrappers outright. It also wouldn't change the write format of the RPCs, since these are already represented as lists on the wire anyhow.